### PR TITLE
Just a few dullahans fixes. And no, it doesn't fix headless monkys yet.

### LINKED ISSUE #32230: Reset perspective breaks examining
Title explains it, lost the ability to examine anything except by right click examining after switching my view between my head and body as a dullahan. Relog fixed this issue.

### LINKED ISSUE #32255: Dullahan explodes if thrown into recycler, being unable to reenter brain
Recycler wasn't emagged.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Just fixing few issues with the buggiest out of the loop species of them all, whose existence contradicts so many lines of bodyparts and organs code in so many ways.
More or less the same changes I did for another server months ago (minus the part enabling them to space-up safely without hardsuit helmet).

## Why It's Good For The Game
This will close #32230, close #32255, also fixing dullahans being unable to receive visible/hearable messages and some issues with their regenerate_limbs and revive.

## Changelog
:cl:
fix: Dullahans won't gib when their head is thrown in a recycler, ahealed or have their limbs regenerated somehow.
fix: Dullahans' heads can now be picked up.
fix: Fixes dullahans being unable to receive visible/hearable messages (excluding chat messages)
/:cl:

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -262,6 +262,7 @@
 		var/atom/A = processing_list[1]
 		if(A.flags_1 & HEAR_1)
 			. += A
+			SEND_SIGNAL(A, COMSIG_ATOM_HEARER_IN_VIEW, processing_list, .)
```
Comment (Ghommie) [inline code/__HELPERS/game.dm]: A little concerned about possible performance drawbacks here, but this is so far the easiest and least hacky solution I had.

**Comment by Ghommie** [comment]
Spurious asset cache runtime struck Travis again.

File: code/modules/mob/living/carbon/human/species_types/dullahan.dm [hunk 2]
```diff
@@ -24,17 +24,20 @@
 
 /datum/species/dullahan/on_species_gain(mob/living/carbon/human/H, datum/species/old_species)
 	. = ..()
-	H.flags_1 &= ~HEAR_1
+	DISABLE_BITFIELD(H.flags_1, HEAR_1)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/human/species_types/dullahan.dm]: This macro is bad, don't use it

File: code/modules/mob/living/carbon/human/species_types/dullahan.dm [hunk 3]
```diff
@@ -24,17 +24,20 @@
...
+			head.throwforce = 25
+			myhead = new /obj/item/dullahan_relay (head, H)
+			H.put_in_hands(head)
+			var/obj/item/organ/eyes/E = H.getorganslot(ORGAN_SLOT_EYES)
+			var/datum/action/item_action/organ_action/dullahan/D = locate() in E?.actions
+			D?.Trigger()
 
 /datum/species/dullahan/on_species_loss(mob/living/carbon/human/H)
-	H.flags_1 |= ~HEAR_1
+	ENABLE_BITFIELD(H.flags_1, HEAR_1)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/human/species_types/dullahan.dm]: Ditto

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/__HELPERS/game.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/__HELPERS/game.dm]: Don't worry about it, signals cost less than proc overhead with no listeners

File: code/modules/mob/living/carbon/human/species_types/dullahan.dm (hunk 2 above)
Comment (Ghommie) [reply code/modules/mob/living/carbon/human/species_types/dullahan.dm]: Not gonna lie, I saw this coming. Yet I hate tildes.

**Pull request merged**